pkg/sql/create_table.go

```diff
@@ -185,6 +185,21 @@ func (n *createTableNode) startExec(params runParams) error {
 	}
 	if isTemporary {
 		telemetry.Inc(sqltelemetry.CreateTempTableCounter)
+
+		// TODO(#46556): support ON COMMIT DROP and DELETE ROWS on TEMPORARY TABLE.
+		// If we do this, the n.n.OnCommit variable should probably be stored on the
+		// table descriptor.
+		// Note UNSET / PRESERVE ROWS behave the same way so we do not need to do that for now.
+		switch n.n.OnCommit {
+		case tree.CreateTableOnCommitUnset, tree.CreateTableOnCommitPreserveRows:
+		default:
+			return errors.AssertionFailedf("ON COMMIT value %d is unrecognized", n.n.OnCommit)
```

Comment:
isn't one of these cases a candidate for an uimplemented error?

Reply:
i'm open to either way - the way i see it to get here you must delete the error from the parser, or make the parser allow arbitrary value.

Reply:
gotcha. I'm fine with this.
(although now I could see us returning an unimplemented error in the parser)

Reply:
I'd prefer handling the given cases in the parser and returning unimplemented errors. There are only a few behaviors here, why do we need to handle arbitrary string input?

Reply:
We aren't handling it here - this is a guard against someone accidentally putting a new enum. And forgetting to do the TODO here.

Reply:
The Parser will error on all expected cases.

Reply:
Sorry, i misunderstood the discussion here.